doc/guides/device_mgmt/ota.rst

```diff
@@ -0,0 +1,83 @@
+.. _ota:
+
+Over-the-Air Update
+###################
+
+Overview
+********
+
+Over-the-Air (OTA) Update is a method for delivering firmware updates to remote
+devices using a network connection. Although the name implies a wireless
+connection, updates received over a wired connection (such as Ethernet)
+are still commonly referred to as OTA updates. This approach requires server
+infrastructure to host the firmware binary and implement a method of signaling
+when an update is available. Security is a concern with OTA updates; firmware
+binaries should be cryptographically signed and verified before flashing.
+
+The :ref:`dfu` section discusses upgrading Zephyr firmware using MCUboot. The
+same method can be used as part of OTA. The binary is first downloaded
+into an unoccupied code partition, then flashed using the `MCUboot bootloader`_
+process.
+
+Examples of OTA
+***************
+
+Golioth
+=======
+
+`Golioth`_ is an IoT management platform that includes OTA updates. Devices are
+configured to observe your available firmware revisions on the Golioth Cloud.
+When a new version is available, the device downloads and flashes the binary. In
+this implementation, the connection between cloud and device is secured using
+TLS/DTLS, and the signed firmware binary is confirmed by MCUboot before flashing
+occurs.
+
+1. A working sample can be found on the `Golioth Zephyr-SDK repository`_
+2. The `Golioth OTA documentation`_ includes complete information about the
+   versioning process
+
+Eclipse hawkBit |trade|
+=======================
+
+`Eclipse hawkBit`_ |trade| is an update server framework that uses polling on a
+REST api to detect firmware updates. When a new update is detected, the binary
+is downloaded and MCUboot verifies the signature before flashing the firmware.
+
+There is a :ref:`hawkbit-api-sample` included in the
+Zephyr :ref:`mgmt-samples` section.
+
+UpdateHub
+=========
+
+`UpdateHub`_ is a platform for remotely updating embedded devices. Updates can
+be manually triggered or monitored via polling. When a new update is detected,
+the binary is downloaded and MCUboot verifies the signature before flashing the
+firmware.
+
+There is an :ref:`updatehub_fota_sample` included in the Zephyr
+:ref:`mgmt-samples` section.
+
+SMP Server
+==========
+
+A Simple Management Protocol (SMP) server can be used to update firmware via
+Bluetooth Low Energy (BLE) or UDP. :ref:`mcu_mgr` is used to send a signed firmware
+binary to the remote device where it is verified and flashed by MCUboot.
+
+There is an :ref:`smp_svr_sample` included in the Zephyr :ref:`mgmt-samples`
+section.
+
+Lightweight M2M (LWM2M)
+=======================
+
+The :ref:`lwm2m_interface` protocol includes support for firmware update via
+``CONFIG_LWM2M_FIRMWARE_UPDATE_OBJ_SUPPORT``. Devices securely connect to an
```

Comment:
```suggestion
:kconfig:`CONFIG_LWM2M_FIRMWARE_UPDATE_OBJ_SUPPORT`. Devices securely connect to an
```

When referring to Kconfig options within the Zephyr documentation, it's preferred to use the `:kconfig:` Sphinx role like the above. This generates a link to a page with more information about the option, such as its dependencies, help text, etc.

(Similarly, when referring to devicetree compatibles, use the `:dtcompatible:` role to get a link into the appropriate bindings index page.)

Reply:
I'm glad you mentioned this. When I added this section I first tried using the `:kconfig:` identifier. It generates the link but the url looks like it doesn't go where you think it should:
```
~/golioth-compile/zephyr/doc/_build/html/reference/kconfig/dummy-syms.html#std-kconfig-CONFIG_LWM2M_FIRMWARE_UPDATE_OBJ_SUPPORT
```

I'll put it in and submit an updated commit. Thanks!